integrations/datadog/client.py

```diff
@@ -233,6 +303,173 @@ async def get_slo_history(
             )
             return {}
 
+    async def get_tags(self) -> dict[str, Any]:
+        url = f"{self.api_url}/api/v1/tags/hosts"
+        result = await self._send_api_request(url)
+
+        tags = result.get("tags")
+        return tags
+
+    def _create_query_with_values(
+        self, metric_query: str, variable_values: dict[str, str]
+    ) -> str:
+        """
+        Creates a Datadog query by substituting template variables with their values.
+
+        Args:
+            metric_query (str): The original query string containing template variables (e.g., "avg:container.cpu.usage{$service}").
+            variable_values (dict): A dictionary mapping variable names to their values (e.g., {"service": "svc_name", "env": "prod"}).
+
+        Returns:
+            str: The modified query with variables replaced by their values.
+        """
+        logger.debug(f"Creating query with values: {metric_query} | {variable_values}")
+        for variable_name, value in variable_values.items():
+            metric_query = metric_query.replace(
+                f"${variable_name}", f"{variable_name}:{value}"
+            )
+        return metric_query
+
+    def extract_metric_name(self, query_string: str) -> str | None:
+        """Extracts the Datadog metric name from a query string.
+
+        Args:
+            query_string: The Datadog query string.
+
+        Returns:
+            The metric name, or None if no metric name is found.
+
+        Examples:
+            - "sum:container.memory.usage" -> "container.memory.usage"
+            - "avg:system.cpu.used{service:my-service}" -> "system.cpu.used"
+        """
+        match = re.search(r"(?:\w+:)?(\w+\.\w+\.\w+)", query_string)
```

Comment:
why we are doing that? as eventually I do care about the `avg` in `avg:system.cpu.used`? 
is there a different naming terminology between `avg:system.cpu.used` to `system.cpu.used` in datadog?

Reply:
this was just metadata info. we still have the full metric name in the title of the entity and query property


```
{
  "identifier": "avg:system.mem.used/service:mayicard-mayicore/env:staging",
  "title": "avg:system.mem.used{service:mayicard-mayicore,env:staging}",
  "icon": null,
  "blueprint": "datadogServiceMetric",
  "team": [],
  "properties": {
    "query": "avg:system.mem.used",
    "series": [],
    "res_type": "time_series",
    "from_date": "2024-08-16T09:15:34Z",
    "to_date": "2024-08-16T09:25:34Z",
    "metric_name": "system.mem.used",
    "env": "staging"
  },
  "relations": {
    "service": "mayicard-mayicore"
  }
}
```